AudioLibrary.WMME/WaveStream.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.IO;
+using AudioLibrary.WMME.Native;
+
+namespace AudioLibrary.WMME
+{
+    internal class WaveStream : Stream, IDisposable
+	{
+		private Stream m_Stream;
+		private long m_DataPos;
+		private long m_Length;
+
+		private WAVEFORMATEX m_Format;
+
+		public WAVEFORMATEX Format
+		{
+			get { return m_Format; }
+		}
+
+		private string ReadChunk(BinaryReader reader)
+		{
+			byte[] ch = new byte[4];
+			reader.Read(ch, 0, ch.Length);
+			return System.Text.Encoding.ASCII.GetString(ch);
+		}
+
+		private void ReadHeader()
+		{
+			BinaryReader Reader = new BinaryReader(m_Stream);
+			if (ReadChunk(Reader) != "RIFF")
+				throw new Exception("Invalid file format");
+
+			Reader.ReadInt32(); // File length minus first 8 bytes of RIFF description, we don't use it
+
+			if (ReadChunk(Reader) != "WAVE")
+				throw new Exception("Invalid file format");
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: 'System.Exception' should not be thrown by user code. ([csharpsquid:S112](https://sonarcloud.io/organizations/alex-yevtushenko3820/rules#rule_key=csharpsquid%3AS112))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=IPP&pullRequest=7&issues=AWfG1-gy8DroMTOoScW1&open=AWfG1-gy8DroMTOoScW1)

[ISSUE_KEY]: AWfG1-gy8DroMTOoScW1